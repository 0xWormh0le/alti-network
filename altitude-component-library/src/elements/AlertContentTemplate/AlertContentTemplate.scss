@import '../../styles/_shared.scss';

.FlaviusAlert {
  padding: 0 0 0 8px !important;
  border-radius: 4px;
  box-shadow: $ALERT_BOX_SHADOW;
  overflow: hidden;
  max-width: $ALERT_MAX_WIDTH;

  &--error {
    background-color: $TOMATO !important;
  }

  &--warning {
    background-color: $GOLDEN_YELLOW !important;
  }

  &--info {
    background-color: $AZURE !important;
  }

  &--success {
    background-color: $PALE_GREEN !important;
  }

  &__body {
    background-color: $ALERT_BG;
    color: $DARK_GREY_BLUE;
    padding: $BASE_SPACING $BASE_SPACING * 1.5;
    @include set-typography('BODY', 'normal');
    display: flex;
    align-items: center;
  }

  &__icon {
    margin-right: $BASE_SPACING * 1.5;
  }

  &__content {
    flex: 1;
    @include set-typography('LABEL', 'medium');
  }

  &__description {
    @include set-typography('BODY', 'normal');
  }

  &__close {
    cursor: pointer;
    margin-left: $BASE_SPACING * 1.5;
    outline: none !important;
    color: $BLUE_GREY;
  }

  &__close-icon {
    width: 10px;
    height: 10px;
  }
}
