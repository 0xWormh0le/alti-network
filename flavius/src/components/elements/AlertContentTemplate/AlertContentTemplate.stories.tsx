import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import { AlertContentTemplate, AlertContentTemplateProps } from './AlertContentTemplate'

export default {
  title: 'Elements/AlertContentTemplate',
  component: AlertContentTemplate
} as Meta

const Template: Story<AlertContentTemplateProps> = (args) => <AlertContentTemplate {...args} />

export const Error = Template.bind({})
Error.args = {
  message: 'Message',
  condition: 'error',
  customFields: {
    description: 'Description'
  }
}

export const Warning = Template.bind({})
Warning.args = {
  message: 'Message',
  condition: 'warning',
  customFields: {
    description: 'Description'
  }
}

export const Success = Template.bind({})
Success.args = {
  message: 'Message',
  condition: 'success',
  customFields: {
    description: 'Description'
  }
}

export const Info = Template.bind({})
Info.args = {
  message: 'Message',
  condition: 'info',
  customFields: {
    description: 'Description'
  }
}
