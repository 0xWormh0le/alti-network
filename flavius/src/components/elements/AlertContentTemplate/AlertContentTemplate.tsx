import React from 'react'
import cn from 'classnames'

import IconClose from 'icons/close'
import IconNotificationError from 'icons/notification-error'
import IconNotificationInfo from 'icons/notification-info'
import IconNotificationSuccess from 'icons/notification-success'
import IconNotificationWarning from 'icons/notification-warning'
import './AlertContentTemplate.scss'

export interface AlertContentTemplateProps {
  classNames: string
  condition: 'error' | 'info' | 'success' | 'warning'
  handleClose: () => any
  id: string
  message: string
  styles: React.CSSProperties
  customFields?: { [key: string]: string }
}

const icons = {
  error: IconNotificationError,
  info: IconNotificationInfo,
  success: IconNotificationSuccess,
  warning: IconNotificationWarning
}

export const AlertContentTemplate: React.FC<AlertContentTemplateProps> = ({
  handleClose,
  message,
  condition,
  id,
  styles,
  classNames,
  customFields: { description } = {}
}) => {
  const Icon = icons[condition]
  return (
    <div className={cn(classNames, 'FlaviusAlert', `FlaviusAlert--${condition}`)} id={id} style={styles}>
      <div className='FlaviusAlert__body'>
        <Icon className='FlaviusAlert__icon' />
        <div className='FlaviusAlert__content'>
          {message}
          {description && <div className='FlaviusAlert__description'>{description}</div>}
        </div>
        <span tabIndex={0} className='FlaviusAlert__close' onClick={handleClose}>
          <IconClose className='FlaviusAlert__close-icon' />
        </span>
      </div>
    </div>
  )
}

export default AlertContentTemplate
