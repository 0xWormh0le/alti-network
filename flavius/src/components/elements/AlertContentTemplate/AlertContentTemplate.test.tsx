import React from 'react'
import { render } from '@testing-library/react'

import noop from 'lodash/noop'
import { AlertContentTemplate, AlertContentTemplateProps } from './AlertContentTemplate'

describe('AlertContentTemplate', () => {
  it('renders correctly', () => {
    const props = {
      handleClose: noop,
      message: 'Login successful',
      condition: 'info',
      id: 'test-alert',
      styles: {},
      classNames: 's-alert',
      customFields: { description: 'You logged in successfully as test user' }
    } as AlertContentTemplateProps
    const { container } = render(<AlertContentTemplate {...props} />)
    expect(container).toMatchSnapshot()
  })
})
