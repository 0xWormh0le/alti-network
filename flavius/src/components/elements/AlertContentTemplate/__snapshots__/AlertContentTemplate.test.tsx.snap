// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertContentTemplate renders correctly 1`] = `
<div>
  <div
    class="s-alert FlaviusAlert FlaviusAlert--info"
    id="test-alert"
  >
    <div
      class="FlaviusAlert__body"
    >
      <svg
        class="FlaviusAlert__icon"
        height="16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm0 4c.6 0 1 .4 1 1s-.4 1-1 1-1-.4-1-1 .4-1 1-1zm2 8H6v-1h1V8H6V7h3v4h1v1z"
          fill="#008EED"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="FlaviusAlert__content"
      >
        Login successful
        <div
          class="FlaviusAlert__description"
        >
          You logged in successfully as test user
        </div>
      </div>
      <span
        class="FlaviusAlert__close"
        tabindex="0"
      >
        <svg
          class="FlaviusAlert__close-icon"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.25 1.25l17.5 17.5m0-17.5l-17.5 17.5"
            fill="none"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="4"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
